### PR TITLE
More conventional schematic & lower capacities to confirm to the USB maximum

### PR DESCRIPTION
**Pull request by con-f-use**
The USB specifications state a total maximum of 10uF of decoupling between VUSB and GND. One should leave a little head-room for devices connected to the tinyusbbord's headers, hence 4.7uF should be ok. The ATMega8 doesn't need as much decoupling. Also 100nF instead of 1uF should provide faster immediate decoupling. Further the schematic didn't fully comply to usual conventions (e.g. usually VCC points upwards and GND downwards, junctions don't indicate labled connections etc.). BTW thanks to Stephan for your awesome project.

### CODE REVIEW

**Comment by baerwolf** [comment]
Hi and thx for your feedback.
I will revise it asap. (There are also some new extensions/shields for the tinyUSBboard coming.)

The big cap was just ensurance the PCB will work (since it is/was manufactured in bigger quantities).
(Originally 47uF were planned. But because of higher prices and bad availability they got replaced by 22uF Samsung MLCC in the current Rev.4 release.)
Of course it could be switched out with some smaller MLCC, but I think a bit more capacitance is quite helpful since users tend to connect also motors directly to the board.
However USB usually tolerates this. At least the limited versions (or versions of concerned makers) have an extra polyfuse limiting startup current. 
With the .1uF you are right, they are usually faster since they are available as single layered ones.
Here I also wanted to donate a bit more capacity since the traces are very small.
I took care to get 1uF as single layered X7R ones, too. But I admit I didn't evaluate the difference to .1uF.
So basically I could switch the caps for .1uF in future kits / the schematic - no big deal.
But I think this should be based on some evaluation: Perhaps somebody could be perusated to do this?

Promise, last paragraph for today:
Perhaps you may want to tell me your address, so I can thank you with some Rev. 4 kit or other prototypes - in case you like to be in the loop of ongoing ideas using tinyUSBboard.
(I very much would like to stay in touch.)

BR Stephan

**Comment by con-f-use** [comment]
Hi Stephan,

thanks, that is a generous offer. I'd be happy with a board or two (without components). I have everything here in my workshop (except of the mega8 but I have mega328p). Address is either:

_Edit: addresses removed since this is public._

whatever is better for you. Of course I will pay for the boards.

But concerning the capacitors again: Do not use anything larger than 10uF. That is the maximum, the USB standard allows for bus-powered devices. The poly fuse doesn't have enough resistance, the standard factors in poor and long cables. Anything more can throw off some USB hubs (I had it happen on me with 47uF, took forever to find the problem) or can stress the connection. 4.7uF should be enough in any case.

Cheers,
Christoph

**Comment by baerwolf** [comment]
Hi again.

I will prepare it during the upcoming weekend, preferring the German address.
AND: NO PAYMENT - it is a gift.
You also can tell me what else you may need.

I can imagine the 22uF probably could in rare and unlikly conditions cause some trouble during connect.
Mostly because of jiff short current during charge up (when no polyfuse is there to limit it).
However I have trouble believing/understanding what harm it takes after beeing fully charged up.
It basically is transparent to DC then. Do you have any references to support your claim?
I have seen so many much poorer designed USB devices - some even with dc-dc boost converters having much higher capacity.

I really would like to know much more about this.

So thx again and gn8 for now.

BR Stephan

**Comment by con-f-use** [comment]
From the [USB Compliance Checklist](http://www.usb.org/developers/compliance/check_list/compchkhub080205.pdf) on the [USB Implementer's Forum's Website](http://usb.org):

> [To comply to USB standards] Does the hub limit the inrush current 
> drawn from V BUS , either by using capacitors smaller than 10μF 
> or by using soft-start circuits, such that no more than 10μF of 
> capacitance is charged [...]

The USB standard and that doesn't explain why. I only have anecdotal evidence like this:

I had problems with another design using a hub: The port worked fine until you plugged in the device. After that it stopped working and one needed to kill the hub's power for the port to be working again. I switched the 47uF on the device for a 4.7uF and it worked ever since. I didn't investigate any further. 

I believe some higher quality USB hubs use electronic fuses to react swiftly to over-current conditions and protect your hardware (something a polyfuse can't as they usually have trip times in the ms and trip currents about twice their rating). These fuses are so fast, that by design, they already trip at the inrush of a relatively small capacitor such as a 47uF. I guess that's why the standard specifies 10uF so manufacturers have a value they can plan for.

Interestingly the minimal decoupling capacity is specified as 1uF. I don't expect there to be trouble with 22uF and only in 1% of all use cases with 47uF, but if there's a specification, why not design for it?

**Comment by baerwolf** [comment]
Hi and thx for explaining.

This is very exciting - esp. since you still might have this usb hub for testing.
I will send you an usual kit plus some .1uF and 10uF caps. I also will put a 47uF MLCC of same dimension in there.
Perhaps you want sth. else, too? 

> but if there's a specification, why not design for it?

Yes, that is a valid question. And also fixing is very simple since no different PCB is required,

I guess from the experience with the previous revisions I decided to push the capacity as high as possible.
(As previously mentioned for motors or WS2812B directly attached.)
I guess I missed the USB spec there or didn't care since the board never would get USB certified logo anyway? (Because of V-USB already violating USB spec.)
Also I think the USB spec is a bit harsh here: 1uF to 10uF total capacity. I don't think there is any consumer USB passing that restriction. ;-)
Also tests from all over the world (and I really mean that: USA, Australia, Japan, Philippines, India....) never complained over issues with the USB interface. So I assumed it ok.

But your request istotally ok.
So I suggest to change the schematic to put a note regarding this in there.
(For example like it is done with the green LED resistor: 1k or 10k.)

 BR Stephan

**Comment by tinyusbboard** [comment]
Hi
Sorry it took me so long.
I recently had a look at your changes. 
Some were pretty nice, like for example reordering the USB or having Vcc and GND jut out the connector's pin array.
But with some changes I can't agree: Using extra elemets to lable line endings. It makes the area crowed and the lables hard to read. Raspberry Pi schematics also do it without.

I also saw, there are changes to the board: Can you explain what you did there ?

BR Stephan

**Comment by con-f-use** [comment]
The change in the board file was just an **eagle hiccup**. For some reason eagle kept deleting a trace after the changes in the schematic. The two pads retained their air wires, but I had to redraw the trace exactly as it was before. There is **no change in the copper**. The trace in question was the short one between R3A and R3 on the bottom layer. I used eagle version 7.1.0. I assume there was an implicit change in net name or an eagle but. Bottom line, without it, the forward backward annotation was broken for no reason.

As for the Xref type labels vs. junctions. I guess that's mostly personal taste. I just want to argue, that Xref labels have the **very purpose** of indicating a connection to some other place on the schematic. That's what they are for, while using junctions might be optically more pleasing but IMHO **missuses junctions**. But ultimately I'm not religious about, I just like them better. An alternative might be to alternate the labels and make them a bit bigger like this:

Btw. most of the time I find it better to **avoid labels altogether and just run wires** across the schematic. It looks a bit like a rat nest at first, but guides the eye to the location of the connected pin, that could otherwise be anywhere on the schematic. That shortens the search time. I.e. the other alternative, direct connection everywhere:


I'll change the fork to whatever you want in terms of labels, just consider the alternatives. Then I'll make another pull request. Does that sound good?

(**Also the board arrived. Thank you very much!** I'm on vacation the coming two weeks, but when I get back to my lab in Austria, I'll test it with the USB hub in question.)

**Comment by tinyusbboard** [comment]
Ok.
Thank you for staying on it.
I suggest doing all changes stepwise, with commits for each change.
But we can talk, when you are back: Have a nice and releaxing vacation.

BR Stephan
